hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/rsql/RSQLUtility.java

```diff
@@ -161,6 +167,109 @@ private static Node parseRsql(final String rsql) {
         }
     }
 
+
+    private static final class ValidationRSQLVisitor<A extends Enum<A> & FieldNameProvider> implements RSQLVisitor<Boolean, String>{
+
+        private Class<A> enumType;
+
+        public ValidationRSQLVisitor(final Class<A> enumType){
+            this.enumType = enumType;
+        }
+
+        @Override
+        public Boolean visit(AndNode node, String param) {
+            for(Node child: node.getChildren()){
+                if(!child.accept(this, param))
+                    return false;
+            }
+            return true;
+        }
+
+        @Override
+        public Boolean visit(OrNode node, String param) {
+            for(Node child: node.getChildren()){
+                if(child.accept(this, param))
+                    return true;
+            }
+            return false;
+        }
+
+        @Override
+        public Boolean visit(ComparisonNode node, String param) {
+            try {
+                final A fieldName = getFieldEnumByName(node);
+                getAndValidatePropertyFieldName(fieldName, node);
+            } catch (final RSQLParameterUnsupportedFieldException | IllegalArgumentException e) {
+                return false;
+            }
+            return true;
+        }
+
+        private A getFieldEnumByName(final ComparisonNode node) {
+            String enumName = node.getSelector();
+            final String[] graph = getSubAttributesFrom(enumName);
+            if (graph.length != 0) {
+                enumName = graph[0];
+            }
+            LOGGER.debug("get field identifier by name {} of enum type {}", enumName, enumType);
+            return Enum.valueOf(enumType, enumName.toUpperCase());
+        }
+
+        private static String[] getSubAttributesFrom(final String property) {
+            return property.split("\\" + FieldNameProvider.SUB_ATTRIBUTE_SEPERATOR);
+        }
+
+        private String getAndValidatePropertyFieldName(final A propertyEnum, final ComparisonNode node) {
+
+            final String[] graph = getSubAttributesFrom(node.getSelector());
+
+            validateMapParameter(propertyEnum, node, graph);
+
+            // sub entity need minium 1 dot
+            if (!propertyEnum.getSubEntityAttributes().isEmpty() && graph.length < 2) {
+                throw new RSQLParameterUnsupportedFieldException();
```

Comment:
I wouldn't just throw the plain RSQLParameterUnsupportedFieldException, because in that case user will not be able to understand where the problem occurred, using #createRSQLParameterUnsupportedException gives more information, however you could also improve upon that

Reply:
Done! Using #createRSQLParameterUnsupportedException method now.